lib/atom.cpp

```diff
@@ -112,24 +112,36 @@ Atom::Atom(int Z, double m, int A, NuclearRadiusModel radius_model, double fc,
   // Define radius
   if (A == -1) {
     R = -1;
-  } else {
+  } else if (radius < 0) {
     switch (radius_model) {
       case POINT:
         R = -1;
         break;
       case SPHERE:
+        R = sphereNuclearModel(Z, A);
+        LOG(INFO) << "SPHERE: R extracted as " << R << "\n";
+        break;
       case FERMI2:
         R = sphereNuclearModel(Z, A);
+        LOG(INFO) << "FERMI2: R extracted as " << R << "\n";
         break;
       default:
         R = -1;
         break;
     }
+  } else {
+    R = radius * Physical::fm;
+  }
+
+  if (radius_model==POINT) {
+    R = -1;
+  } else {
+    LOG(INFO) << "R = " << radius << " fm\n";
   }
 
   if (radius_model == FERMI2) {
     V_coulomb = new CoulombFermi2Potential(Z, R, A);
-  } else {
+  } else if (radius_model == SPHERE) {
```

Comment:
Currrenlty if the radiusl_model == 'POINT'  case is ignored, It would be cause error if the keyword `nuclear_model: POINT`.
```suggestion
  } else {
```